docusaurus/docs/recipes/redux.mdx

```diff
@@ -0,0 +1,407 @@
+
+How to use your store and tweak it further:
+- Add your own reducers to the `reducers` object. The `app` reducer above is just a sample reducer.
+- Add your own middlewares to the `middleware` array.
+- The `modules` object allows you to pass arbitrary data to your action creators and middlewares. An API client module is a good example of what `modules` can be used for. Feel free to rename this variable to better suit your needs.
+
+#### `src/shared/state/persist/index.js`
+
+```js
+import { combineReducers } from 'redux';
+import { persistReducer, persistStore } from 'redux-persist';
+import AsyncStorage from '@react-native-community/async-storage';
+import config from './config';
+
+export const createPersistReducer = (reducers) => persistReducer(
+    { key: 'root', storage: AsyncStorage, ...config.root },
+    combineReducers(
+        Object.entries(reducers).reduce((reducers, [key, reducer]) => ({
+            ...reducers,
+            [key]: config[key] ?
+                persistReducer({ ...config[key], key, storage: AsyncStorage }, reducer) :
+                reducer,
+        }), {}),
+    ),
+);
+
+export const createStorePersistor = (store) => persistStore(store);
+```
+
+#### `src/shared/state/persist/config.js`
+
+```js
+import { createMigrate } from 'redux-persist';
+import migrations from './migrations';
+
+export default {
+    root: {
+        debug: __DEV__,
+        version: Object.keys(migrations).pop() || 0,
+        blacklist: [],
+        transforms: [],
+        migrate: createMigrate(migrations, { debug: __DEV__ }),
+    },
+};
+```
+
+#### `src/shared/state/persist/migrations.js`
+
+```js
+export default {};
+```
+
+Persistance allows you to:
+- Select, through blacklists, whitelists and transforms, which parts of the state get to be serialized to disk and deserialized (rehydrated) back to the app.
+- Maintain different versions of the state and move from an older version to a newer one with migrations.
+
+By default, the whole state tree is persisted. Head to [`redux-persist`](https://github.com/rt2zz/redux-persist) repository to learn more about how Redux store persistance works and how it can be configured and customized.
+
+## 3. Create `StoreProvider`
+
+The next step is to put together Redux's `Provider` and `redux-persist`'s `PersistGate`.
+
+Create the files as shown in the sections below.
+
+#### `src/shared/modules/react-native-store/StoreProvider.js`
+
+```js
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from 'react-redux';
+import { PersistGate } from 'redux-persist/integration/react';
+
+const StoreProvider = ({ children, ...props }) => {
+    const { current: { store, purge, persist } } = useRef(props);
+    const [isPurging, setPurging] = useState(purge && !!store.persistor);
+    const childrenFn = (storeReady) => children({ storeReady });
+
+    useEffect(() => {
+        if (!persist) {
+            store?.persistor.pause();
```

Comment:
If you're setting the store prop as required in the PropTypes, why null check it here?

Reply:
The `store` is being required but `persistor` is not. However, we're not really expecting to receive a store without persistence capabilities. This is just a safe guard to ensure an error is thrown if persistence is disabled by the developer and the store cannot persist state. This is effectively a no-op.

Reply:
I think we're missing a `?.` though between `persistor` and `pause()`.

Reply:
Yes, the `?.` should be between the `persistor` and `pause` 😄